src/main.rs

```diff
@@ -85,17 +54,46 @@ fn main() {
         LevelFilter::INFO
     };
     tracing_subscriber::fmt().with_max_level(log_level).init();
-
     tracing::info!("Starting app and waiting for Tendermint to connect...");
 
+    // instantiate the application with a KV store implementation of choice
     let app = BaseCoinApp::new(InMemoryStore::default()).expect("Failed to init app");
-    let app_copy = app.clone();
-    let grpc_port = opt.grpc_port;
-    let grpc_host = opt.host.clone();
-    std::thread::spawn(move || grpc_serve(app_copy, grpc_host, grpc_port));
 
+    // instantiate modules and setup inter-module communication (if required)
+    let auth = Auth::new(app.module_store(&prefix::Auth {}.identifier()));
+    let bank = Bank::new(
+        app.module_store(&prefix::Bank {}.identifier()),
+        auth.account_reader().clone(),
+        auth.account_keeper().clone(),
+    );
+    let ibc = Ibc::new(app.module_store(&prefix::Ibc {}.identifier()));
+    let staking = Staking::new(app.module_store(&prefix::Staking {}.identifier()));
+
+    // register modules with the app
+    let app = app
+        .add_module(prefix::Auth {}.identifier(), auth.clone())
```

Comment:
How about we change `Identifiable::identifier()` to be an associated function, so that we don't need to needlessly clear an object. Would become:

`.add_module(prefix::Auth::identifier(), auth.clone()`